sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorClient.cs

```diff
@@ -176,630 +123,292 @@ public virtual Pageable<string> GetMetricDimensionValues(string metricId, string
+        /// Adds a <see cref="MetricFeedback"/>.
         /// </summary>
-        /// <param name="metricId">The unique identifier of the <see cref="DataFeedMetric"/>.</param>
-        /// <param name="options">The set of options used to configure the request's behavior.</param>
+        /// <param name="feedback">The <see cref="MetricFeedback"/> to be created.</param>
         /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
-        /// <returns>An <see cref="AsyncPageable{T}"/> containing the collection of <see cref="MetricSeriesData"/>.</returns>
-        /// <exception cref="ArgumentNullException"><paramref name="metricId"/> or <paramref name="options"/> is null.</exception>
-        /// <exception cref="ArgumentException"><paramref name="metricId"/> is empty or not a valid GUID.</exception>
-        public virtual AsyncPageable<MetricSeriesData> GetMetricSeriesDataAsync(string metricId, GetMetricSeriesDataOptions options, CancellationToken cancellationToken = default)
+        /// <returns>
+        /// A <see cref="Response{T}"/> containing the result of the operation. The result is a <see cref="MetricFeedback"/>
+        /// containing information about the newly added feedback.
+        /// </returns>
+        /// <exception cref="ArgumentNullException"><paramref name="feedback"/> is null.</exception>
+        public virtual async Task<Response<MetricFeedback>> AddFeedbackAsync(MetricFeedback feedback, CancellationToken cancellationToken = default)
         {
-            Argument.AssertNotNull(options, nameof(options));
+            Argument.AssertNotNull(feedback, nameof(feedback));
 
-            Guid metricGuid = ClientCommon.ValidateGuid(metricId, nameof(metricId));
-            IEnumerable<IDictionary<string, string>> series = options.SeriesKeys.Select(key => key.Dimension);
-            MetricDataQueryOptions queryOptions = new MetricDataQueryOptions(ClientCommon.NormalizeDateTimeOffset(options.StartsOn), ClientCommon.NormalizeDateTimeOffset(options.EndsOn), series);
+            using DiagnosticScope scope = ClientDiagnostics.CreateScope($"{nameof(MetricsAdvisorClient)}.{nameof(AddFeedback)}");
+            scope.Start();
 
-            async Task<Page<MetricSeriesData>> FirstPageFunc(int? pageSizeHint)
+            try
             {
-                using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(MetricsAdvisorClient)}.{nameof(GetMetricSeriesData)}");
-                scope.Start();
+                RequestContent content = MetricFeedback.ToRequestContent(feedback);
+                RequestContext context = new RequestContext()
+                {
+                    CancellationToken = cancellationToken,
+                };
+                Response response = await CreateMetricFeedbackAsync(content, context).ConfigureAwait(false);
+
+                var location = response.Headers.TryGetValue("Location", out string value) ? value : null;
```

Comment:
If `location` is set to `null`, what exception is thrown?  Is it the same as in the HLC?

Reply:
Nothing chaged in this experiment, it is same as HLC. If `location` is set to null, it throws [ArgumentException](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/ClientCommon.cs#L144).